lib/words_report.rb

```diff
@@ -0,0 +1,25 @@
+class WordsReport
+  attr_reader :words
+
+  # Arbitrary definition of popular
+  POPULAR_COUNT = 5
+
+  def initialize(counts)
+    @counts = counts
+    @words = words_by_frequency
```

Comment:
Looking at it again, I was setting @words inside of words_by_frequency, and then using its return value. That was sub-optimal. Removing that made @counts really stick out.

Changing it to a utility function that doesn't mess with the object's state is much clearer at that point.